pkg/os/volume/api.go

```diff
@@ -170,18 +174,24 @@ func (VolumeAPI) ResizeVolume(volumeID string, size int64) error {
 	} else {
 		finalSize = size
 	}
+	klog.V(4).Infof("finalSize for resize of the volume: %s is %d ", volumeID, finalSize)
 
 	currentSize, err := getVolumeSize(volumeID)
 	if err != nil {
 		return fmt.Errorf("error getting the current size of volume (%s) with error (%v)", volumeID, err)
 	}
-
+	klog.V(4).Infof("currentSize of the volume: %s is %d ", volumeID, currentSize)
 	//if the partition's size is already the size we want this is a noop, just return
 	if currentSize >= finalSize {
 		klog.V(2).Infof("Attempted to resize volume %s to a lower size, from currentBytes=%d wantedBytes=%d", volumeID, currentSize, finalSize)
 		return nil
 	}
-
+	// Resize-Partition requires minimum extent size 1 MB
+	// if extent size for resize partition is less than 1 MB, skip expanding volume
+	if finalSize-currentSize < MinimumExpandSize {
```

Comment:
While the error is handled by CSI Proxy it'd be great to let the user know what happened and that can only happen if CSI Proxy throws the error and the CSI Drivers handles it.

Reply:
do you mean return an error with the code using https://github.com/grpc/grpc-go/blob/master/status/status.go#L62-L65 and handle this error in the driver to make a decision about failure or success?

```goalng
	if finalSize-currentSize < MinimumExpandSize {
		return status.Errorf(codes.OutOfRange, 
			"minimum extent size is 1 MB. Skip resize for volume %s from currentBytes=%d to wantedBytes=%d ", volumeID, currentSize, finalSize)
	}
```

Reply:
`codes.OutOfRange` is a good idea, yes I think that the driver should handle the error, if no resize happened then it can log that too without erroring for better visibility rather than checking the CSI Proxy logs

Reply:
addressed review comment with this commit - https://github.com/kubernetes-csi/csi-proxy/pull/196/commits/603b06b028cdc2b1dc9e1c657d4294b2d8d59df4